runtime/lib/capi/ExecutionContext.hpp

```diff
@@ -153,6 +218,24 @@ class ExecutionContext final {
 
             return true;
         }
+
+        // TODO: Once all devices are shared libraries, they all need to be loaded.
+        // During this transition period, there are several ways in which we can do this.
+        // This try catch is just for allowing the previous mechanism to still succeed
+        // while keeping the implementation of SharedLibraryManager as a minimal as possible.
+        // Once all devices are shared libraries, we can replace initDevice with loadDevice.
+        //
+        // Yes, I know there is a performance impact. But this try-catch will be removed once
+        // all devices are shared libraries.
+        try {
+            QuantumDevice *impl = loadDevice(std::string(name));
```

Comment:
If it's always called with a lvalue, then you can further import the code by removing this cast from `std::string_view` to `std::string`.

Reply:
I cannot use std::string_view because I need a C string . There is no conversion from std::string_view to a C string except `std::string_view::data()` but it is not guaranteed to be to have a terminating null pointer.

Reply:
Added it. Thanks!